dynamo/preprocessing/Preprocessor.py

```diff
@@ -225,22 +225,19 @@ def standardize_adata(self, adata: AnnData, tkey: str, experiment_type: str) ->
 
         adata.uns["pp"] = {}
         adata.uns["pp"]["norm_method"] = None
+        self.basic_stats(adata)
         self.add_experiment_info(adata, tkey, experiment_type)
         main_info_insert_adata("tkey=%s" % tkey, "uns['pp']", indent_level=2)
-        main_info_insert_adata("experiment_type=%s" % experiment_type, "uns['pp']", indent_level=2)
+        main_info_insert_adata("experiment_type=%s" % adata.uns["pp"]["experiment_type"], "uns['pp']", indent_level=2)
+
         main_info("making adata observation index unique...")
-        self.unique_var_obs_adata(adata)
         self.convert_layers2csr(adata)
+        self.collapse_species_adata(adata)
 
-        if self.collapse_species_adata:
-            main_info("applying collapse species adata...")
-            self.collapse_species_adata(adata)
-
-        if self.convert_gene_name:
-            main_info("applying convert_gene_name function...")
-            self.convert_gene_name(adata)
-            main_info("making adata observation index unique after gene name conversion...")
-            self.unique_var_obs_adata(adata)
```

Comment:
we should actually keep this. because we don't always need to run collapse_species_adata and convert_gene_name. We only need to do this if we have e.g. uu, ul, su, sl 4 species and the gene name is not in official name but emsemble id

Reply:
in fact, we need to to have a function to directly detect whether it is necessary to collapse or convert gene names automatically

Reply:
I am uncertain if the current 'if statements' are necessary since the functions being called already contain condition checking statements. 
For example,
if np.all([name in adata.layers.keys() for name in splicing_and_labeling]):   -----> if we have e.g. uu, ul, su, sl 4 species 
if np.all(adata.var_names.str.startswith("ENS")) or scopes is not None:      ------> the gene name is not in official name but emsemble id

In my opinion, we can remove the if statements and call each function, including the unique_var_obs_adata to clarify and simplify the code. However, if you believe that those statements are necessary or a redundant function to check is needed, I am open to creating them.